PostHog/posthog
# SAML support

## Changes

Contributes to #5647. Introduces base support for SAML login / account provisioning. Option A, instance-based, single tenant.

With this PR users will be able to set up instance-based SAML through env var config (full docs coming + examples with Okta & OneLogin). Works only on self-hosted.

Full backend E2E tests included (you can [use this OneLogin tool](https://www.samltool.com/decode.php) to decode the test payloads if you're curious)

**How to test?**
Easiest method is through Okta developer account.
1. Set [environment variables](https://posthog.slack.com/archives/C01G8S5T08Z/p1629951397000800)
2. Log in to Okta (credentials on 1Password)
3. See the magic happen!

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)
